version: '3.7'

services:
  db:
    networks:
      - app_network
    image: mysql:8.0
    restart: always
    env_file:
      - .env
    volumes:
      - db_data:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_DATABASE=${DATABASE_NAME}
      - DATABASE_USER=${DATABASE_USER}
    ports:
      - '3306:3306'
    healthcheck:
      test: [ "CMD-SHELL", "mysqladmin ping -h localhost -uroot -p${MYSQL_ROOT_PASSWORD}" ]
      interval: 10s
      timeout: 5s
      retries: 5

  web:
    networks:
      - app_network
    build: .
    command: python /code/manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
      - media_data:/code/media
    ports:
      - '8000:8000'
    depends_on:
      db:
        condition: service_healthy
    env_file:
      - .env
    logging:
      driver: json-file
      options:
        max-size: "10m"
        max-file: "3"
    environment:
      - DB_HOST=db
      - DB_PORT=3306
      - DB_NAME=${DATABASE_NAME}
      - DB_USER=${DATABASE_USER}
      - DB_PASS=${MYSQL_PASSWORD}
      - DJANGO_SECRET_KEY=${DOCKER_COMPOSE_DJANGO_SECRET_KEY}
      - DJANGO_DEBUG=${DOCKER_COMPOSE_DJANGO_DEBUG}
      - DJANGO_ZIBAL_MERCHANT_ID=${DOCKER_COMPOSE_DJANGO_ZIBAL_MERCHANT_ID}

volumes:
  db_data:
  media_data:

networks:
  app_network:
